src/main/java/com/frostwire/jlibtorrent/TorrentStats.java

```diff
@@ -0,0 +1,179 @@
+ * on the speed in which the results of the speed of the Alert event
+ * listener is retrieved, which can have a delay in milliseconds calculated
+ * in avg of 1-90ms.
+ *
+ * @author haperlot
+ */
+
+public final class TorrentStats {
+
+
+    private SessionManager sessionManager;
+    private TorrentHandle torrentHandle;
+    private long samplingIntervalInMs;
+    private long maxHistoryInMs;
+    public static final String UPLOAD = "UPLOAD";
+    public static final String DOWNLOAD = "DOWNLOAD";
+    private Queue<Integer> downloadRate = new LinkedList<Integer>();
+    private Queue<Integer> uploadRate = new LinkedList<Integer>();
+    private long tStart; //for collecting sampling interval time
+
+
+    public TorrentStats(final SessionManager sessionManager, TorrentHandle torrentHandle, long samplingIntervalInMs, long maxHistoryInMs) {
+        this.sessionManager = sessionManager;
+        this.torrentHandle = torrentHandle;
+        this.samplingIntervalInMs = samplingIntervalInMs;
+        this.maxHistoryInMs = maxHistoryInMs;
+        this.tStart = System.currentTimeMillis();
+
+        startAlertListener();
+    }
+
+
+    /**
+     * It will start the alert event listener, as soon as the sampling interval
+     * meets the requested time it will save the values on queues. If the queues
+     * reach max size denoted by the maxHistory time, they will poll() the head
+     * element, in our case the oldest inserted.
+     */
+    private void startAlertListener() {
+
+        sessionManager.addListener(new AlertListener() {
+            @Override
+            public int[] types() {
+                return null;
+            }
+
+            @Override
+            public void alert(Alert<?> alert) {
+                AlertType type = alert.type();
+
+                switch (type) {
+
+                    case TORRENT_ADDED:
+                        System.out.println("Torrent added ");
+                        ((TorrentAddedAlert) alert).handle().resume();
+                        break;
+
+                    case BLOCK_FINISHED:
+                        //only true if samplingIntervalInMs,as soon as I get it from the alert margin of error 5-90 ms.
+                        if ((System.currentTimeMillis() - tStart) >= samplingIntervalInMs) {
+                            tStart = System.currentTimeMillis();
+
+                            //if !paused && !finished, it could be paused during the download
+                            if (!torrentHandle.status().isFinished() && !torrentHandle.status().isPaused()) {
+
+                                downloadRate.add(torrentHandle.status().downloadRate());
+                                uploadRate.add(torrentHandle.status().uploadRate());
+
+                                //if the sampling # exceeded the limit, remove head
+                                if (downloadRate.size() * samplingIntervalInMs > maxHistoryInMs) {
+                                    downloadRate.poll();
```

Comment:
I think you should abstract the data structure with a better interface/name and move out the call to `poll()`. Is `LinkedList` an efficient double linked list?